src/main/java/io/strimzi/kafka/metrics/DataPointSnapshotBuilder.java

```diff
@@ -11,21 +11,32 @@
 import io.prometheus.metrics.model.snapshots.PrometheusNaming;
 import io.prometheus.metrics.model.snapshots.Quantiles;
 import io.prometheus.metrics.model.snapshots.SummarySnapshot;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  * Builder methods to convert Kafka metrics into Prometheus metrics
  */
 public class DataPointSnapshotBuilder {
 
+    private static final Logger LOG = LoggerFactory.getLogger(DataPointSnapshotBuilder.class.getName());
+
     /**
      * Create a datapoint for a {@link InfoSnapshot} metric
      * @param labels The labels associated with the datapoint
      * @param value The value to insert as a label
      * @param metricName The name of the new label
      * @return The {@link InfoSnapshot.InfoDataPointSnapshot} datapoint
      */
+
     public static InfoSnapshot.InfoDataPointSnapshot infoDataPoint(Labels labels, Object value, String metricName) {
-        Labels newLabels = labels.add(PrometheusNaming.sanitizeLabelName(metricName), String.valueOf(value));
+        String newLabelName = PrometheusNaming.sanitizeLabelName(metricName);
+        Labels newLabels = labels;
+        if (labels.contains(newLabelName)) {
+            LOG.warn("Ignoring metric value duplicate key: {} from metric: {}",  newLabelName, metricName);
```

Comment:
I wonder if it would be worth checking the value of the existing label and only print the warning if the value is different from `value`. WDYT?

Can we also improve the message? We're not ignoring the metric value, we're unable to add a new label.

Reply:
Refactored. Is the new message ok?